ebean-core/src/main/java/io/ebeaninternal/api/SpiDdlGenerator.java

```diff
@@ -8,8 +8,14 @@
   /**
    * Generate and run the DDL for drop-all and create-all scripts.
    * <p>
-   * Run based on on property settings for ebean.ddl.generate and ebean.ddl.run etc.
+   * Run based on on property settings for ebean.ddl.generate
    */
-  void execute(boolean online);
+  void generateDdl();
 
+  /**
+   * Runs the created DDL files.
+   * <p>
+   * Run based on on property settings for ebean.ddl.run etc.
+   */
+  void runDdl();
```

Comment:
So both of these methods `runDdl()` `runMigration()` are actually not what I was expecting.  Instead I was expecting `start()` (along the lines of the conversation that took place in the prior issue that was just closed).

There are a number of thoughts which suggest `start()` is still the way to go:
1)  runDdl() should really only be a test concern - we never expect to runDdl in production, only migrations are run in production. 
2) `start` does other things like register with ebean container for clustering support

So at this stage I think this PR isn't what we want. I think we still want `start()` (for programmatic start of the ebean database instance, which could fail due to database/datasource not ready, and could run the migration based on existing config.

I can see that `start()` by itself doesn't support the for loop other tenants, running the migration for each tenant case. Hmmm.

I don't see the change that moves the cluster registration that happens in DefaultContainer to only execute after successful `start()`.  To me we need to sort out `start()` first.

Reply:
Hello @rbygrave I've updated this PR and made the `start()` method part of the API.

Migration/Ddl will run if start() is invoked.

We now need also a `skipStart` in the ServerConfig. I found no other property that may be used here, as `dbOffline` means "really offline" - so no ClusterManager etc will be started.

`start()` is also skipped, if `tenantMode.isDdlEnabled()` is false 

In a multi tenant environment you must call `start()` for each tenant. 

With this PR you can cover the following use cases
- create ebean server, when DB is not yet available, but it is expected to come online later
- do DbMigration (and DdlRun) for each tenant`

I'm not yet sure, if `start()` is the correct term here. I think about
- server.init(); // makes more sense, when iterating over tenants
- config.skipInit
- TenantMode.supportsInit

I would like to give you an outlook on my further ideas: 
1.) DdlGenerator becomes a Plugin: Please take a look at this commit: https://github.com/FOCONIS/ebean/commit/699f7a8449d09c84f73c0a4730a5ba31e0430b1b - this would allow to remove the SpiDdlGenerator completely

2.) DbMigration becomes also a Plugin: Please take a look at these commits in this branch: https://github.com/FOCONIS/ebean/commits/DbMigrationPlugin

Both code-branches are experimental but may affect further decisions. First I want to get this PR discussed & eventually merged

cheers
Roland

Reply:
I don't think they can become plugins due to ordering ... the ddl ought to run before any/all "plugins" are run etc